application/src/main/java/org/gxf/soapbridge/application/factories/HttpsUrlConnectionFactory.java

```diff
@@ -0,0 +1,104 @@
+      final SslContextCacheService sslContextCacheService,
+      final HostnameVerifierFactory hostnameVerifierFactory) {
+    this.sslContextCacheService = sslContextCacheService;
+    this.hostnameVerifierFactory = hostnameVerifierFactory;
+  }
+
+  /**
+   * Create an {@link HttpsURLConnection} instance for the given arguments.
+   *
+   * @param uri The full URI of the end-point for this connection.
+   * @param host The host consists of domain name, server name or IP address followed by the port.
+   *     Example: localhost:443
+   * @param contentLength The content length of the SOAP payload which will be sent to the end-point
+   *     using this connection.
+   * @param commonName The common name for the organization.
+   * @return Null in case the {@link SSLContext} cannot be created or fetched from the {@link
+   *     SslContextCacheService}, or a configured and initialized {@link HttpsURLConnection}
+   *     instance.
+   * @throws UnableToCreateHttpsURLConnectionException In case the configuration and/or
+   *     initialization of an {@link HttpsURLConnection} instance fails.
+   */
+  public HttpsURLConnection createConnection(
+      final String uri, final String host, final String contentLength, final String commonName)
+      throws UnableToCreateHttpsURLConnectionException {
+    try {
+      // Get SSLContext instance.
+      final SSLContext sslContext;
+      if (StringUtils.hasText(commonName)) {
+        sslContext = sslContextCacheService.getSslContextForCommonName(commonName);
+      } else {
+        sslContext = sslContextCacheService.getSslContext();
+      }
+      // Check SSLContext instance.
+      if (sslContext == null) {
+        LOGGER.error(
+            "SSLContext instance is null. Unable to create HttpsURLConnection instance for uri: {}, host: {}, content length: {}, common name: {}",
+            uri,
+            host,
+            contentLength,
+            commonName);
+        return null;
+      }
+      // Create connection.
+      HttpsURLConnection.setDefaultHostnameVerifier(hostnameVerifierFactory.getHostnameVerifier());
+      final HttpsURLConnection connection = (HttpsURLConnection) new URL(uri).openConnection();
+      connection.setSSLSocketFactory(sslContext.getSocketFactory());
+      connection.setDoInput(true);
+      connection.setDoOutput(true);
+      connection.setRequestMethod("POST");
+      connection.setRequestProperty(
+          "Accept-Encoding", "text/xml;charset=" + StandardCharsets.UTF_8.name());
+      connection.setRequestProperty("Accept-Charset", StandardCharsets.UTF_8.name());
+      connection.setRequestProperty(
+          "Content-Type", "text/xml;charset=" + StandardCharsets.UTF_8.name());
+      connection.setRequestProperty("SOAP-ACTION", "");
+      connection.setRequestProperty("Content-Length", contentLength);
+      connection.setRequestProperty("Host", host);
+      connection.setRequestProperty("Connection", "Keep-Alive");
```

Comment:
Ik gebruik voor de property keys liever velden dan magic strings.
`org.apache.http.HttpHeaders` of `org.springframework.http.HttpHeaders` heeft de meeste ook. Maar we kunnen ook eigen constanten maken.